src/test/java/com/cag/cagbackendapi/controllers/UserControllerTest.kt

```diff
@@ -242,4 +241,83 @@ class UserControllerTest {
         verifyNoMoreInteractions(validationService, userService)
     }
 
+    @Test
+    fun deleteUser_validInput_200OK() {
+        val testAuthKey = "testAuthKey"
+        val userId = "123e4567-e89b-12d3-a456-426614174000"
+        val userUUID = UUID.fromString(userId)
+        val userData = UserResponseDto(userUUID, "John", "Smith", "johnjohn@aol.com", true, null)
+
+        doNothing().whenever(validationService).validateAuthKey(testAuthKey)
+        whenever(userService.deleteUser(userId)).thenReturn(userData)
+
+        userController.deleteUser(testAuthKey, userId)
+
+        verify(validationService).validateAuthKey(testAuthKey)
+        verify(userService).deleteUser(userId)
+        verifyNoMoreInteractions(validationService, userService)
+    }
+
+    @Test
+    fun deleteUser_missingUserId_400BadRequest(){
+        val testAuthKey = "testAuthKey"
+        val userId = "test user"
+
+        val badRequestException = BadRequestException(DetailedErrorMessages.INVALID_USER_ID, null)
+
+        doNothing().whenever(validationService).validateAuthKey(testAuthKey)
+        whenever(userService.deleteUser(userId)).thenThrow(badRequestException)
+
+        val actual = assertThrows<BadRequestException> {
+            userController.deleteUser(testAuthKey, userId)
+        }
+
+        assertEquals(actual.message, badRequestException.message)
+
+        verify(validationService).validateAuthKey(testAuthKey)
+        verify(userService).deleteUser(userId)
+        verifyNoMoreInteractions(validationService, userService)
+    }
+
+    @Test
+    fun deleteUser_invalidAuthKey_401Unauthorized(){
+        val testAuthKey = "testAuthKey"
+        val userId = "test user"
+
+        val unauthorizedException = UnauthorizedException(DetailedErrorMessages.MISSING_AUTH_KEY, null)
+
+        whenever(validationService.validateAuthKey(testAuthKey)).thenThrow(unauthorizedException)
+
+        val actual = assertThrows<UnauthorizedException> {
+            userController.deleteUser(testAuthKey, userId)
+        }
+
+        assertEquals(actual.message, unauthorizedException.message)
+
+        verify(validationService).validateAuthKey(testAuthKey)
+        verifyZeroInteractions(userService)
+    }
+
+
+    //REVIEW THIS TEST
+    @Test
+    fun deleteUser_invalidUserId_404UserNotFound(){
+        val testAuthKey = "testAuthKey"
+        val userId = "nonUserUUID"
```

Comment:
The userId you're using isn't a valid UUID but it doesn't matter since the controller doesn't validate for a valid UUID. The real problem is that you created a BadRequestException and threw that instead of creating a NotFoundException. Since we're mocking, it will throw whatever exception you tell it to throw - even if the UUID you provide doesn't exist. The point of the test is to check that the controller handles the exception correctly when It gets it